modules/python/scene/source/scene.i

```diff
@@ -91,3 +91,51 @@
       $self->mParams[idx] = val;
     }
 };
+
+
+%extend scene::ProjectionModel
+{
+  /* Extend the Python interface to imageToScene to take a sequence of  */
+  /* row indices and a sequence of column indices and return a list of  */
+  /* projected points                                                   */
+  PyObject* imageToScene(PyObject* row_seq,
+                         PyObject* col_seq,
+                         math::linear::VectorN<3,double> ground_ref_point,
+                         math::linear::VectorN<3,double> ground_plane_normal)
+  {
+    if (!PySequence_Check(row_seq))
+    {
+      PyErr_SetString(PyExc_TypeError,"Expecting a sequence of row indices.");
+      return NULL;
+    }
+    if (!PySequence_Check(col_seq))
+    {
+      PyErr_SetString(PyExc_TypeError,"Expecting a sequence of column indices.");
+      return NULL;
+    }
+    int N = PyObject_Length(row_seq);
```

Comment:
Changed ‘int’ to ‘Py_ssize_t’ throughout update.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Tuesday, November 24, 2015 11:38 AM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: David Starinshak <David.Starinshak@mdaus.com>
Subject: Re: [six-library] Extend projection model python interface (#32)


In modules/python/scene/source/scene.i<http://cp.mcafee.com/d/5fHCNEp6hAgdEIcfcLcTvpdTdIIczxNEVvvdLzCm6kjhOqejqdTHLe6zCVEVudEEEFLLcECzBwSxFCi7kDIpgEjQG2BkdrFOVJl3mWsKrgojd7b3Xz_nVOWq9EVWZOWqrOf8IELETVqWdAklrL8YG7DR8OJMddECQPt-jjsuovsp7e6XCOsVHkiP2cFASO7bV0zF7WOxfai3OVYy4tymNpIY_6kMcgBiPtaiH0lyM-evd78VNwT2Ivmfb0EzapdIxOVJ-WdPobZ8Qg6BKAwYKv8x7oBziWq81EfDY43h0J6EzZFOH0QgaRgYYQgb5FAWMd44Ej-bKSYQg6yw36kPhirsodC_h_KxxiFEdA>:

> +  PyObject* imageToScene(PyObject* row_seq,

> +                         PyObject* col_seq,

> +                         math::linear::VectorN<3,double> ground_ref_point,

> +                         math::linear::VectorN<3,double> ground_plane_normal)

> +  {

> +    if (!PySequence_Check(row_seq))

> +    {

> +      PyErr_SetString(PyExc_TypeError,"Expecting a sequence of row indices.");

> +      return NULL;

> +    }

> +    if (!PySequence_Check(col_seq))

> +    {

> +      PyErr_SetString(PyExc_TypeError,"Expecting a sequence of column indices.");

> +      return NULL;

> +    }

> +    int N = PyObject_Length(row_seq);

Use Py_ssize_t here and below

—
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/1jWVIidEIcfcLcTvpdTdIIczxNEVvvdLzCm6kjhOqejqdTHLe6zCVEVudEEEFLLcECzBwSxFCi7kDIpgEjQG2BkdrFOVJl3mWsKrgojd7b3Xz_nVOWq9EVWZOWqrOf8IELETVqWdAklrL8YG7DR8OJMddFCQPt-jjsuovsp7e6XCOsVHkiP2cFASO7bV0zF7WOxfai3OVYy4tymNpIY_6kYtAvZ0cyM-evd78VNwT2Ivmfb0EzapdIxOVJ-WdPobZ8Qg6BKAwYKv8x7oBziWq81EfDY43h0J6EzZFOH0QgaRgYYQgb5FAWMd44Ej-bKSYQg6yw36kPhirsodJTBB0FX>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.